### PR TITLE
add port vcpkg-tool-python2

### PR DESCRIPTION
**Pull request by Neumann-A**
adds two ports
- vcpkg-tool-lessmsi to not use msiexec to extract msi files
- vcpkg-tool-python2 to extract python2 with lessmsi.

Other stuff:
- make qt5-base switch to python3 (needs ci testing)
- make qt5-webengine use python2 explicitly (similar to qtbase/qtwebengine)
- added the vcpkg-tool-python2 to some ports using python2 (but not all)

### CODE REVIEW

File: ports/qt5-webengine/portfile.cmake [hunk 1]
```diff
@@ -15,6 +15,10 @@ if(NOT VCPKG_TARGET_IS_WINDOWS)
 endif()
 include(${CURRENT_INSTALLED_DIR}/share/qt5/qt_port_functions.cmake)
 
+vcpkg_find_acquire_program(PYTHON2)
```
Comment (Osyotr) [inline ports/qt5-webengine/portfile.cmake:18]: It's already there :)

File: ports/vcpkg-tool-lessmsi/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,17 @@
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
+
+set(version v1.10.0)
+
+vcpkg_download_distfile(archive_path
+    URLS "https://github.com/activescott/lessmsi/releases/download/${version}/lessmsi-${version}.zip"
+    FILENAME "lessmsi-${version}"
```
Comment (strega-nil-ms) [inline ports/vcpkg-tool-lessmsi/portfile.cmake:7]:
```suggestion
    FILENAME "lessmsi-${version}.zip"
```

File: ports/vcpkg-tool-lessmsi/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    SHA512 91be9363d75e8ca0129304008ddc26fe575cc4fd76d7f43ef0a6ff414855dc1c6e412f4e694b2950026e02cc3d31b18bd8c2e4c03e1ddce01477f3f2d2197479
+)
+
+file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
+
+vcpkg_execute_in_download_mode(
+                    COMMAND "${CMAKE_COMMAND}" -E tar xzf "${archive_path}"
+                    WORKING_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/${PORT}"
+                )
```
Comment (strega-nil-ms) [inline ports/vcpkg-tool-lessmsi/portfile.cmake:16]:
```suggestion
file(ARCHIVE_EXTRACT
    INPUT "${archive_path}"
    DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}"
)
```

File: ports/vcpkg-tool-lessmsi/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,17 @@
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
```
Comment (strega-nil-ms) [inline ports/vcpkg-tool-lessmsi/portfile.cmake:1]: I'd like a script to go along with this eventually, but I'm not going to block on that.

File: ports/vcpkg-tool-python2/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                    ERROR_FILE "lessmsi-${arg_FILENAME}-err.log"
+                    RESULT_VARIABLE error_code
+                )
+    if(error_code)
+        message(FATAL_ERROR "Couldn't extract Python2 with lessmsi!")
+    endif()
+    message(STATUS "Extracting Python2 ... finished!")
+    file(RENAME "${CURRENT_PACKAGES_DIR}/tools/python-2.7.18.amd64/SourceDir/" "${CURRENT_PACKAGES_DIR}/tools/${PORT}/")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/tools/python-2.7.18.amd64")
+    z_vcpkg_find_acquire_program_find_internal("${program}"
```
Comment (strega-nil-ms) [inline ports/vcpkg-tool-python2/portfile.cmake:41]: where does `${program}` come from? also, why is this needed?

File: ports/qt5-webengine/portfile.cmake (hunk 1 above)
Comment (Neumann-A) [reply ports/qt5-webengine/portfile.cmake:18]: I see I forgot to commit my latest changes.

File: ports/vcpkg-tool-python2/portfile.cmake (hunk 5 above)
Comment (Neumann-A) [inline ports/vcpkg-tool-python2/portfile.cmake:41]:
```suggestion
    z_vcpkg_find_acquire_program_find_internal("PYTHON2"
```

Comment (Neumann-A) [reply ports/vcpkg-tool-python2/portfile.cmake:41]: Just to make sure that Python2 will be found by `vcpkg_find_acquire_program`.

File: ports/vcpkg-tool-python2/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,58 @@
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
+
+if(VCPKG_TARGET_IS_WINDOWS)
```
Comment (Neumann-A) [inline ports/vcpkg-tool-python2/portfile.cmake:3]:
```suggestion
set(arch_prefix "")
if(VCPKG_TARGET_IS_WINDOWS)
```

File: ports/vcpkg-tool-python2/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    set(program_name python)
+    set(program_version 2.7.18)
+    if (VCPKG_TARGET_ARCHITECTURE STREQUAL x86)
+        set(tool_subdirectory "python-${program_version}-x86")
+        set(download_urls "https://www.python.org/ftp/python/${program_version}/python-${program_version}.msi")
+        set(download_filename "python-${program_version}.msi")
+        set(download_sha512 2c112733c777ddbf189b0a54047a9d5851ebce0564cc38b9687d79ce6c7a09006109dbad8627fb1a60c3ad55e261db850d9dfa454af0533b460b2afc316fe115)
+    else()
+        set(tool_subdirectory "python-${program_version}-x64")
+        set(download_urls "https://www.python.org/ftp/python/${program_version}/python-${program_version}.amd64.msi")
```
Comment (Neumann-A) [inline ports/vcpkg-tool-python2/portfile.cmake:13]:
```suggestion
        set(arch_suffix ".amd64")
        set(download_urls "https://www.python.org/ftp/python/${program_version}/python-${program_version}.amd64.msi")
```

File: ports/vcpkg-tool-python2/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+                    WORKING_DIRECTORY "${output_path}" 
+                    OUTPUT_FILE "${CURRENT_BUILDTREES_DIR}/lessmsi-${TARGET_TRIPLET}-out.log"
+                    ERROR_FILE "${CURRENT_BUILDTREES_DIR}/lessmsi-${TARGET_TRIPLET}-err.log"
+                    RESULT_VARIABLE error_code
+                )
+    if(error_code)
+        message(FATAL_ERROR "Couldn't extract Python2 with lessmsi!")
+    endif()
+    message(STATUS "Extracting Python2 ... finished!")
+    file(RENAME "${output_path}/python-2.7.18.amd64/SourceDir/" "${output_path}/${PORT}/")
```
Comment (Neumann-A) [inline ports/vcpkg-tool-python2/portfile.cmake:39]:
```suggestion
    file(RENAME "${output_path}/python-2.7.18${arch_suffix}/SourceDir/" "${output_path}/${PORT}/")
```

File: ports/vcpkg-tool-python2/portfile.cmake [hunk 9]
```diff
@@ -0,0 +1,58 @@
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
+
+if(VCPKG_TARGET_IS_WINDOWS)
+    set(program_name python)
```
Comment (Neumann-A) [inline ports/vcpkg-tool-python2/portfile.cmake:4]:
```suggestion
    set(arch_suffix "")
    set(program_name python)
```

**Comment by JackBoosY** [comment]
Since my PR https://github.com/microsoft/vcpkg/pull/23478 and this PR requires each other, can you please merge my changes into this?

**Comment by JackBoosY** [comment]
And what about the python3:x64-linux regression?

**Comment by Neumann-A** [comment]
> And what about the python3:x64-linux regression?

that is a vcpkg-tool update regression. Previously stuff like that was not checked on other platforms.
Maybe @Hoikas can help here?

**Comment by JackBoosY** [comment]
> > And what about the python3:x64-linux regression?
> 
> that is a vcpkg-tool update regression. Previously stuff like that was not checked on other platforms then windows. Maybe @Hoikas can help here?

Ah I remember that just now, that's because the dependency libraries are mismatched.

**Comment by JackBoosY** [comment]
```
CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:696 (message):
Unable to find PYTHON2
```

**Comment by Neumann-A** [comment]
> can you please merge my changes into this?
@JackBoosY : please do that yourself. I want have time until very much later today.

> CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:696 (message):
Unable to find PYTHON2

Where did you see that?

**Comment by JackBoosY** [comment]
> Where did you see that?

When building `vcpkg-find-acquire-program:x64-windows`:
```
Starting package 7/7: vcpkg-find-acquire-program:x64-windows
Building package vcpkg-find-acquire-program[core]:x64-windows...
-- Installing port from location: C:\a\1\s\scripts/test_ports\vcpkg-find-acquire-program
-- Downloading https://dl.google.com/go/go1.16.6.windows-386.zip -> go1.16.6.windows-386.zip...
-- Downloading https://download.qt.io/official_releases/jom/jom_1_1_3.zip;https://mirrors.ocf.berkeley.edu/qt/official_releases/jom/jom_1_1_3.zip -> jom_1_1_3.zip...
-- Downloading https://www.nasm.us/pub/nasm/releasebuilds/2.15.05/win32/nasm-2.15.05-win32.zip;https://fossies.org/windows/misc/nasm-2.15.05-win32.zip -> nasm-2.15.05-win32.zip...
-- Downloading https://strawberryperl.com/download/5.32.1.1/strawberry-perl-5.32.1.1-32bit.zip -> strawberry-perl-5.32.1.1-32bit.zip...
-- Downloading https://www.tortall.net/projects/yasm/snapshots/v1.3.0.6.g1962/yasm-1.3.0.6.g1962.exe -> yasm-1.3.0.6.g1962.exe...
-- Installing: D:/downloads/tools/yasm/1.3.0.6/yasm.exe
-- Downloading https://www.python.org/ftp/python/2.7.18/python-2.7.18.amd64.msi -> python-2.7.18.amd64.msi...
The Windows Installer Service could not be accessed. This can occur if the Windows Installer is not correctly installed. Contact your support personnel for assistance.

CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:696 (message):
  Unable to find PYTHON2
Call Stack (most recent call first):
  scripts/test_ports/vcpkg-find-acquire-program/portfile.cmake:5 (vcpkg_find_acquire_program)
  scripts/ports.cmake:145 (include)


Error: Building package vcpkg-find-acquire-program:x64-windows failed with: BUILD_FAILED
Elapsed time for package vcpkg-find-acquire-program:x64-windows: 11.77 s

```

**Comment by Neumann-A** [comment]
@JackBoosY: Yeah that is because I didn't adjust all usages of `vcpkg_find_acquire_program(PYTHON2)` to depend on `vcpkg-tool-python2`. Basically just the manifest needs to include it to make it work.

**Comment by JackBoosY** [comment]
@Neumann-A Does that mean we can't use this changes to fix the python2 regression completely?

**Comment by Neumann-A** [comment]
> Does that mean we can't use this changes to fix the python2 regression completely?

It is only fixed if the ports using python2 are depending on `vcpkg-tool-python2` or if  `vcpkg-tool-python2`  gets installed before the port using python2. I didn't fix all ports in this PR which are using python2. The python2 issues is mainly a CI issue not a user issue. So this issues in CI should be fixed if it is observed

**Review by vicroms — DISMISSED** [review]

**Comment by Neumann-A** [comment]
depends #23525 to be green in CI
